include/openssl/core_numbers.h

```diff
@@ -95,12 +96,68 @@ OSSL_CORE_MAKE_FUNC(int, OP_digest_final,
 OSSL_CORE_MAKE_FUNC(int, OP_digest_digest,
                     (const unsigned char *in, size_t inl, unsigned char *out,
                      size_t *out_l))
+
 OSSL_CORE_MAKE_FUNC(void, OP_digest_cleanctx, (void *vctx))
 OSSL_CORE_MAKE_FUNC(void, OP_digest_freectx, (void *vctx))
 OSSL_CORE_MAKE_FUNC(void *, OP_digest_dupctx, (void *vctx))
 OSSL_CORE_MAKE_FUNC(size_t, OP_digest_size, (void))
 OSSL_CORE_MAKE_FUNC(size_t, OP_digest_block_size, (void))
 
+
+/* Symmetric Ciphers */
+
+# define OSSL_OP_CIPHER                              2
+
+# define OSSL_FUNC_CIPHER_NEWCTX                     1
+# define OSSL_FUNC_CIPHER_ENCRYPT_INIT               2
+# define OSSL_FUNC_CIPHER_ENCRYPT_UPDATE             3
+# define OSSL_FUNC_CIPHER_ENCRYPT_FINAL              4
+# define OSSL_FUNC_CIPHER_DECRYPT_INIT               5
+# define OSSL_FUNC_CIPHER_DECRYPT_UPDATE             6
+# define OSSL_FUNC_CIPHER_DECRYPT_FINAL              7
+# define OSSL_FUNC_CIPHER_CIPHER                     8
+# define OSSL_FUNC_CIPHER_FREECTX                    9
+# define OSSL_FUNC_CIPHER_DUPCTX                    10
+# define OSSL_FUNC_CIPHER_KEY_LENGTH                11
+# define OSSL_FUNC_CIPHER_IV_LENGTH                 12
+# define OSSL_FUNC_CIPHER_BLOCK_SIZE                13
+# define OSSL_FUNC_CIPHER_GET_PARAMS                14
+# define OSSL_FUNC_CIPHER_CTX_GET_PARAMS            15
+# define OSSL_FUNC_CIPHER_CTX_SET_PARAMS            16
+
+OSSL_CORE_MAKE_FUNC(void *, OP_cipher_newctx, (void))
+OSSL_CORE_MAKE_FUNC(int, OP_cipher_encrypt_init, (void *vctx,
+                                                  const unsigned char *key,
+                                                  const unsigned char *iv))
+OSSL_CORE_MAKE_FUNC(int, OP_cipher_encrypt_update,
+                    (void *, unsigned char *out, size_t *outl,
+                     const unsigned char *in, size_t inl))
+OSSL_CORE_MAKE_FUNC(int, OP_cipher_encrypt_final,
+                    (void *, unsigned char *out, size_t *outl))
```

Comment:
Similarly, I would like a parameter where the maximum size of the output buffer is given, both in `update` and `final`